### PR TITLE
Np 9238 get person with employments

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java [hunk 1]
```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
         assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_GATEWAY_TIMEOUT));
     }
 
+    @Test
+    void shouldNotReturnEmploymentsWhenNotAuthorized() throws IOException, ApiGatewayException {
+        apiClient = spy(apiClient);
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendQuery(EMPTY_MAP, VALID_PATH_PARAM).getBodyObject(Person.class);
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:203]: Is `sendQuery` a `sendUnauthorizedQuery`?

Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:203]: what is the emptyMap, is it empty QueryParameters?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java [hunk 2]
```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
         assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_GATEWAY_TIMEOUT));
     }
 
+    @Test
+    void shouldNotReturnEmploymentsWhenNotAuthorized() throws IOException, ApiGatewayException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:200]: "shouldNotRequestEmploymentsFromProxiedServerWhenClientIsNotAuthorized" ?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java [hunk 3]
```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
...
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendQuery(EMPTY_MAP, VALID_PATH_PARAM).getBodyObject(Person.class);
+
+        verify(apiClient, times(0)).fetchGetResultWithAuthentication(any());
+        assertThat(actual.getEmployments().size(), equalTo(0));
+    }
+
+    @Test
+    void shouldReturnEmploymentsWhenAuthorized() throws IOException, ApiGatewayException {
+        apiClient = spy(apiClient);
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:211]: Suggestion:
```
 @Test
    void shouldReturnEmploymentsWhenAuthorized() throws IOException, ApiGatewayException {
        setUpApiClientToRespondToPersonAndEmploymentRequestsWhenRequestIsAuthorizedAndPersonHasEmployment();
    
        handler = new FetchCristinPersonHandler(apiClient, environment);
        var actual = sendAuthorizedQuery(EDIT_OWN_INSTITUTION_USERS).getBodyObject(Person.class);

        assertThat(actual.getEmployments().size(), equalTo(EXPECTED_EMPLOYMENT_SIZE_WHEN_AUTHORIZED));
    }

 private void setUpApiClientToRespondToPersonAndEmploymentRequestsWhenRequestIsAuthorizedAndPersonHasEmployment() throws ApiGatewayException {
        apiClient = spy(apiClient);
        var cristinPersonUri = URI.create(EXPECTED_CRISTIN_URI_WITH_IDENTIFIER);
        doReturn(getResponseWithoutEmployment()).when(apiClient).fetchGetResultWithAuthentication(cristinPersonUri);
        var cristinPersonEmploymentsUri = URI.create(EXPECTED_CRISTIN_EMPLOYMENTS_URI_WITH_IDENTIFIER);
        doReturn(getResponseWithOnlyEmployments()).when(apiClient)
            .fetchGetResultWithAuthentication(cristinPersonEmploymentsUri);
    }


```

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java [hunk 4]
```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
...
+        doReturn(getResponseWithOnlyEmployments()).when(apiClient)
+            .fetchGetResultWithAuthentication(cristinPersonEmploymentsUri);
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendAuthorizedQuery(EDIT_OWN_INSTITUTION_USERS).getBodyObject(Person.class);
+
+        assertThat(actual.getEmployments().size(), equalTo(EXPECTED_EMPLOYMENT_SIZE_WHEN_AUTHORIZED));
+    }
+
+    @Test
+    void shouldReturnOkEvenWhenEmploymentsAreEmpty() throws ApiGatewayException, IOException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:224]: do the same here as suggested above

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java [hunk 5]
```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
...
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendAuthorizedQuery(EDIT_OWN_INSTITUTION_USERS);
+        var employments = actual.getBodyObject(Person.class).getEmployments();
+
+        assertThat(actual.getStatusCode(), equalTo(HttpURLConnection.HTTP_OK));
+        assertThat(employments.size(), equalTo(0));
+    }
+
+    @Test
+    void shouldNotReturnEmploymentsWhenNotHavingRequiredAccessRight() throws IOException, ApiGatewayException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java:240]: use the "setUpApiClientToRespondToPersonAndEmploymentRequestsWhenRequestIsAuthorizedAndPersonHasEmployment" method

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Comment by torarnet** [comment]
There will be changes in how employment data is returned from the Cristin Api. With the new changes employment data is included in the original Person GET, meaning we don't have to do an extra request. I am considering closing this PR.

**Review by brinxmat — COMMENTED** [review]
See previous reviewer's comments

**Pull request closed without merging**